### PR TITLE
Bug Fix: Decompilation of Access Rules

### PR DESCRIPTION
**Pull request by 0xOmarA**
## Summary

Previously, `AccessRuleConfig` made use of `BTreeMap`s which does not preserve the order of the items inside it. This lead the manifest decompiler to fail at creating identical manifests. Additionally, compiling these two manifests produced two non-identical compiled manifests which caused issues in the RET.

### CODE REVIEW

File: radix-engine-interface/src/blueprints/resource/access_rules.rs [hunk 1]
```diff
@@ -9,6 +9,8 @@ use sbor::rust::collections::BTreeMap;
 use sbor::rust::str;
 use sbor::rust::string::String;
 use sbor::rust::string::ToString;
+use utils::rust::prelude::index_map;
+use utils::rust::prelude::IndexMap;
```
Comment (dhedey) [inline radix-engine-interface/src/blueprints/resource/access_rules.rs:13]: You could/should just replace all of these with `import sbor::rust::prelude::*` or `import utils::rust::prelude::*`

File: transaction/examples/account/new_complex.rtm [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
...balId("resource_sim1q2atsr8kvzrkdpqe7h94jp9vleraasdw348gn8zg9g6n6g50t6hwlp:[cedebe542c3294b26810b8861e24b7941ae5d4967b926a3f1769]")))))), Tuple(Enum(0u8), "create_proof"), Enum(0u8, Enum(2u8, Enum(0u8, Enum(0u8, Enum(0u8, NonFungibleGlobalId("resource_sim1q2atsr8kvzrkdpqe7h94jp9vleraasdw348gn8zg9g6n6g50t6hwlp:[cedebe542c3294b26810b8861e24b7941ae5d4967b926a3f1769]")))))), Tuple(Enum(0u8), "create_proof_by_ids"), Enum(0u8, Enum(2u8, Enum(0u8, Enum(0u8, Enum(0u8, NonFungibleGlobalId("resource_sim1q2atsr8kvzrkdpqe7h94jp9vleraasdw348gn8zg9g6n6g50t6hwlp:[cedebe542c3294b26810b8861e24b7941ae5d4967b926a3f1769]")))))), Tuple(Enum(0u8), "withdraw_non_fungibles"), Enum(0u8, Enum(2u8, Enum(0u8, Enum(0u8, Enum(0u8, NonFungibleGlobalId("resource_sim1q2atsr8kvzrkdpqe7h94jp9vleraasdw348gn8zg9g6n6g50t6hwlp:[cedebe542c3294b26810b8861e24b7941ae5d4967b926a3f1769]"))))))), Map<String, Enum>(), Enum(0u8, Enum(0u8)), Map<Tuple, Enum>(Tuple(Enum(0u8), "lock_fee"), Enum(0u8, Enum(2u8, Enum(0u8, Enum(0u8, Enum(0u8, NonFungibleGlobalId("resource_sim1q2atsr8kvzrkdpqe7h94jp9vleraasdw348gn8zg9g6n6g50t6hwlp:[cedebe542c3294b26810b8861e24b7941ae5d4967b926a3f1769]"))))))), Map<String, Enum>(), Enum(0u8, Enum(0u8)));
```
Comment (dhedey) [inline transaction/examples/account/new_complex.rtm:2]: Can you parametrize these resource addreses?

Actually if you rebase on top of https://github.com/radixdlt/radixdlt-scrypto/pull/985 it'd probably be easier from a merging perspective :)

File: transaction/src/manifest/e2e.rs [hunk 3]
```diff
@@ -568,6 +568,22 @@ CREATE_ACCOUNT;
...
+    fn test_create_account_complex() {
+        compile_and_decompile_with_inversion_test(
+            "create_account",
+            &apply_replacements_to_manifest(
+                include_str!("../../examples/account/new_complex.rtm").to_string(),
+            ),
+            &NetworkDefinition::simulator(),
+            vec![],
+            r##"
+CREATE_ACCOUNT_ADVANCED
```
Comment (dhedey) [inline transaction/src/manifest/e2e.rs:581]: Ditto RE parametrizing (see other comment)

**Comment by 0xOmarA** [comment]
@dhedey Addressed the comments above and rebased to `feature/vanity-addresses`.

**Comment by dhedey** [comment]
@0xOmarA  - Ace :) - but I can't see the new PR for it - did it go missing? :)

**Comment by 0xOmarA** [comment]
@dhedey After our discussion in Slack we have agreed that we would no longer use `IndexMap`s for `AccessRulesConfig` and instead we would settle for the solution suggested by Yulong. 

Yulong has agreed to take on that PR so I have closed this one.

**Pull request closed without merging**